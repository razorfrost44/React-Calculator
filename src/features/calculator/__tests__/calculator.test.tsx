import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import configureStore from "redux-mock-store"
import { Provider } from "react-redux"
import Calculator from "../calculator"

const mockStore = configureStore([])

describe("Control Pad", () => {
  let store
  let component
  const initialState = {
    currentValue: 0,
    previousValue: 0,
    operator: "",
  }

  const setup = () => {
    store = mockStore({
      calculator: { calculator: initialState },
    })
    component = renderer.create(
      <Provider store={store}>
        <Calculator />
      </Provider>,
    )
  }

  beforeEach(() => {
    setup()
  })

  it("should match snapshot", () => {
    // arrange
    // act
    // assert
    expect(component.toJSON()).toMatchSnapshot()
  })
})
