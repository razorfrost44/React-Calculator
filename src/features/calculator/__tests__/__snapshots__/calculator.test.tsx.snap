// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Control Pad > should match snapshot 1`] = `
<div
  className="calculator"
>
  <section
    className="calculator-display-section"
  >
    <span
      className="previous-value-span"
      data-testid="saved-value"
    />
    <input
      aria-label="Display for the current calculator number"
      className="calculator-display"
      id="calc-display"
      onChange={[Function]}
      type="number"
    />
  </section>
  <section
    className="control-pad"
  >
    <button
      aria-label="Clear"
      className="control-button clear"
      onClick={[Function]}
    >
      CE
    </button>
    <button
      aria-label="Flip Sign"
      className="control-button flipSign"
      onClick={[Function]}
    >
      +/-
    </button>
    <button
      aria-label="Percent"
      className="control-button percent"
      onClick={[Function]}
    >
      %
    </button>
    <button
      aria-label="Divide"
      className="control-button divide last-column"
      onClick={[Function]}
    >
      /
    </button>
    <button
      aria-label="Seven"
      className="control-button seven"
      onClick={[Function]}
    >
      7
    </button>
    <button
      aria-label="Eight"
      className="control-button eight"
      onClick={[Function]}
    >
      8
    </button>
    <button
      aria-label="Nine"
      className="control-button nine"
      onClick={[Function]}
    >
      9
    </button>
    <button
      aria-label="Multiply"
      className="control-button multiply last-column"
      onClick={[Function]}
    >
      X
    </button>
    <button
      aria-label="Four"
      className="control-button four"
      onClick={[Function]}
    >
      4
    </button>
    <button
      aria-label="Five"
      className="control-button five"
      onClick={[Function]}
    >
      5
    </button>
    <button
      aria-label="Six"
      className="control-button six"
      onClick={[Function]}
    >
      6
    </button>
    <button
      aria-label="Subtract"
      className="control-button subtract last-column"
      onClick={[Function]}
    >
      -
    </button>
    <button
      aria-label="One"
      className="control-button one"
      onClick={[Function]}
    >
      1
    </button>
    <button
      aria-label="Two"
      className="control-button two"
      onClick={[Function]}
    >
      2
    </button>
    <button
      aria-label="Three"
      className="control-button three"
      onClick={[Function]}
    >
      3
    </button>
    <button
      aria-label="Sum"
      className="control-button sum last-column"
      onClick={[Function]}
    >
      +
    </button>
    <button
      aria-label="Zero"
      className="control-button zero"
      onClick={[Function]}
    >
      0
    </button>
    <button
      aria-label="Period"
      className="control-button period"
      onClick={[Function]}
    >
      .
    </button>
    <button
      aria-label="Equals"
      className="control-button equals last-column"
      onClick={[Function]}
    >
      =
    </button>
  </section>
</div>
`;
